sql-plugin/src/main/scala/com/nvidia/spark/rapids/AbstractGpuJoinIterator.scala

```diff
@@ -200,53 +203,133 @@ abstract class SplittableJoinIterator(
     val wasInitialJoin = isInitialJoin
     isInitialJoin = false
     if (pendingSplits.nonEmpty || stream.hasNext) {
-      val cb = if (pendingSplits.nonEmpty) {
-        opTime.ns {
-          withResource(pendingSplits.dequeue()) {
-            _.getColumnarBatch()
-          }
-        }
+      val scb = if (pendingSplits.nonEmpty) {
+        pendingSplits.dequeue()
       } else {
         val batch = withResource(stream.next()) { lazyBatch =>
           opTime.ns {
-            lazyBatch.releaseBatch()
+            SpillableColumnarBatch(lazyBatch.releaseBatch(),
```

Comment:
Does it make sense to refactor this code a bit before the call to `splitAndSave`? Since split and save could run OOM.

Reply:
Good point.  We likely need a retry around the split-and-save code, which is also called during a split-and-retry for createGatherer.